### PR TITLE
chore: externalize Vale styles (for the master branch)

### PR DESCRIPTION
**Pull request by themr0c**
## What does this pull request change?

Externalize Vale styles. Use Vale styles maintained in external repositories:

- https://github.com/vale-at-red-hat/vale-at-red-hat/releases/latest/download/RedHat.zip
-  https://github.com/vale-at-red-hat/chedocs/releases/latest/download/CheDocs.zip 

## What issues does this pull request fix or reference?

Single source for the Vale style used for all projects using the RedHat Style Guide, and the Vale style used for other Eclipse Che projects (the blog).

## Specify the version of the product this pull request applies to

7.41.0

## Pull Request checklist

The author and the reviewers validate the content of this pull request with the following checklist, in addition to the [automated tests](code_review_checklist.adoc).

- Any procedure:
  - [ ] Successfully tested.
- Any page or link rename:
  - [ ] The page contains a redirection for the previous URL.
  - Propagate the URL change in:
    - [ ] Dashboard [default branding data](https://github.com/eclipse-che/che-dashboard/blob/main/packages/dashboard-frontend/src/services/bootstrap/branding.constant.ts)
    - [ ] Chectl [constants.ts](https://github.com/che-incubator/chectl/blob/master/src/constants.ts)
- [ ] Builds on [Eclipse Che hosted by Red Hat](https://workspaces.openshift.com).
- [ ] the *`Validate language on files added or modified`* step reports no vale warnings.

### CODE REVIEW

File: .vale.ini [hunk 1]
```diff
@@ -43,17 +37,17 @@ properties = md
...
 
 [*.ini]
-BasedOnStyles = CheDocs,IBM
-IBM.Slash = NO
-IBM.Spacing = NO
-CheDocs.Spelling = NO
+BasedOnStyles = RedHat
+RedHat.Slash = NO
+RedHat.Spacing = NO
+RedHat.Spelling = NO
```
Comment (themr0c) [inline .vale.ini:53]:
```suggestion
RedHat.Spelling = NO
RedHat.url = NO

```

File: .vale.ini [hunk 2]
```diff
@@ -43,17 +37,19 @@ properties = md
...
+RedHat.url = NO
```
Comment (themr0c) [inline .vale.ini:54]:
```suggestion
RedHat.url = NO
RedHat.ConfigMap  = NO
```

File: .vale.ini [hunk 3]
```diff
@@ -43,17 +37,20 @@ properties = md
...
+RedHat.ConfigMap  = NO
```
Comment (themr0c) [inline .vale.ini:55]:
```suggestion
RedHat.ConfigMap  = NO
RedHat.write = NO
```

**Review by benoitf — APPROVED** [review]
I think it might be good to check sha before always downloading the same file when we're launching the command but could be an improvement

Also styles could be cached in github action as well using sha as a key as well to be more robust/faster

**Review by tolusha — APPROVED** [review]

**Comment by themr0c** [comment]
Good ideas @benoitf. Maybe for the next iteration. Let's see first if downloading the styles is consuming too much time.

**Comment by themr0c** [comment]
> I think it might be good to check sha before always downloading the same file when we're launching the command but could be an improvement
> 
> Also styles could be cached in github action as well using sha as a key as well to be more robust/faster

=> https://issues.redhat.com/browse/RHDEVDOCS-3601

**Comment by themr0c** [comment]
https://git-scm.com/ is down today :(

**Pull request merged**